README.md

```diff
@@ -66,11 +66,51 @@ The metadata generated is extremely specific to our use case. In addition, we pe
 
 The metadata generated produces various statistical and geometric properties of a biological specimen image or collection in a JSON format. When a single file is passed, the data is yielded to the console (stdout). When a directory is passed, the data is stored in a JSON file.
 
+### Model
+The trained model is available as "Drexel_metadata_generator" at https://datacommons.tdai.osu.edu/dataverse/fish-traits/.
+The model can be downloaded from that website or via the [dva](https://github.com/Imageomics/dataverse-access) command line utility.
+To download from the command line install dva then run the following command:
+```
+dva download --url https://datacommons.tdai.osu.edu/ doi:10.5072/FK2/MMX6FY .
+```
+The above command will download the file and verify the checksum.
+
+### Running
 To generate the metadata, run the following command:
 ```bash
 pipenv run python3 gen_metadata.py [file_or_dir_name]
 ```
 
+Usage:
+```
+gen_metadata.py [-h] [--device {cpu,cuda}] [--outfname OUTFNAME] [--maskfname MASKFNAME] [--visfname VISFNAME]
+                       file_or_directory [limit]
+```
+
+The `limit` parameter will limit 
+the number of files processed in the directory. The `limit` positional argument is only applicable when passing a directory. 
+
+#### Device Configuration
+By default `gen_metadata.py` requires a GPU (cuda).
+To use a CPU instead pass the `--device cpu` argument to `gen_metadata.py`.
+
+#### Single File Usage
+The following three arguments are only supported when processing a single image file:
+- `--outfname <filename>` - When passed the script will save the output metadata JSON to `<filename>` instead of printing to the console (the default behavior when processing one file).
+- `--maskfname <filename>` - Enables logic to save an output mask to `<filename>` for the single input file.
+- `--visfname <filename>` - Changes the script to save the output visualization to `<filename>` instead of the hard coded location.
+
+These arguments are meant to simplify adding `gen_metadata.py` to a workflow that process files individually.
+
+
+### Running with Singularity
+A Docker container is automatically built for each **drexel_metadata** release. This container has the requirements installed and includes the model file.
+To run the singularity container for a specific version follow this pattern:
+```
+singularity run docker://ghcr.io/hdr-bgnn/drexel_metadata:<release> python gen_metadata.py ...
```

Comment:
Are you sure this shouldn't be `python3` instead of `python`? And the usage command above doesn't prefix with the interpreter, so maybe we don't need to here, either, which would be naturally consistent with the usage command above.